### ISSUE TITLE
Images don't seem to work

### ISSUE THREAD

**Issue by mithro** [SymbiFlow/sphinxcontrib-markdown-symlinks#6]
If I have a file like `abc/X.md` with the contents;
```md
![an image](img.svg)
```

If I link this into `docs/abc.md` and use `sphinxcontrib-markdown-symlinks` the image doesn't work.

**Comment by acomodi** [comment]
@mithro I guess this happens because the `img.svg` is relative to the `abc/X.md` path, and the script cannot handle images relative paths.
I will look into that and fix it.

**Comment by acomodi** [comment]
@mithro So, indeed, the link of the image refers to an non-existent file in the `docs` tree.
A possible way to solve this would be to add a symbolic link to the image in the `docs` tree as it happens for the various `X.md` files (sth similar to this https://github.com/SymbiFlow/prjxray/blob/7a48d1e0e663a2d20a45a05157d4a11531916d86/docs/Makefile#L24-L39).

I will look for a better solution to solve this directly in the `sphinxcontrib-markdown-symlinks` package.

<blockquote><img src="https://repository-images.githubusercontent.com/114421621/68e8c380-e157-11e9-83b1-7c985f136526" width="48" align="right"><div><img src="https://github.githubassets.com/favicon.ico" height="14"> GitHub</div><div><strong><a href="https://github.com/SymbiFlow/prjxray">SymbiFlow/prjxray</a></strong></div><div>Documenting the Xilinx 7-series bit-stream format. - SymbiFlow/prjxray</div></blockquote>

**Comment by mithro** [comment]
Can you just rerewrite the image URL?

**Comment by acomodi** [comment]
@mithro I have looked into this and the issue cannot be solved in the same way as for the xref links. I have tried to apply the same method also for images links, but I think that the solution would be over-complicated.
My suggestion is to actually use the absolute path for the image or to create a script to generate the sym-link of the images in the docs tree.

**Comment by mithro** [comment]
@acomodi Do you mean converting the image link to an absolute path which the HTML builder will then copy the image to the correct location and make it relative again?

**Comment by acomodi** [comment]
@mithro My mistake, I meant the actual URL path of the image in the target repository.
E.g.:
```
![an image](img.svg) --> ![an image](https://github.com/<repo>/<sth>/img.svg)
```

This way the image should be correctly visualized in the resulting html. I need to double check that though.

I think though that the best solution would be to add and use the sym-links to the image in the `code tree`

**Comment by mithro** [comment]
I don't think using the URL path of the image in the target repository is a good idea.

**Comment by acomodi** [comment]
@mithro Ya, I have thought of a possible solution, but I need to test if it works.

The main issue is that, when visiting the image nodes with the recommonmark parser, I cannot treat their URLs / references as the link nodes (e.g. creating `pending_xref` nodes to be evaluated after parsing).

I will try to modify the Parser class to be able to already establish what is the correct link/reference to 
the image and apply one of the following to solutions:
- directly copy/create sym-link to the image in the right location of the docs tree;
- apply the correct link to the image node pointing to the image in the code tree.

**Comment by mithro** [comment]
> I will try to modify the Parser class to be able to already establish what is the correct link/reference to
> the image and apply one of the following to solutions:
> 
> * directly copy/create sym-link to the image in the right location of the docs tree;

I don't think this is the right solution -- the problem is if two markdown files have images of the same name then they will conflict.

If we did do this solution we should;
 (a) Calculate the md5sum / sha1 hash of the image contents.
 (b) symlink <hash>.img to the source image.
 (c) Rewrite the link to <hash>.img

> * apply the correct link to the image node pointing to the image in the code tree.

I think this should be used. Did you try this version?
